File: frontend/stockr-app/src/app/models/product.ts
import { Category } from './category';
import { Supplier } from './supplier';

export interface Product {
  id?: string;
  name: string;
  sku: string;
  description: string;
  categoryId: string;
  supplierId?: string;
  costPrice: number;
  salePrice: number;
  category?: Category;
  supplier?: Supplier;
}

export interface CreateProductRequest {
  name: string;
  sku: string;
  description: string;
  categoryId: string;
  supplierId?: string;
  costPrice: number;
  salePrice: number;
}

export interface UpdateProductRequest {
  id: string;
  name: string;
  sku: string;
  description: string;
  categoryId: string;
  supplierId?: string;
  costPrice: number;
  salePrice: number;
}

File: backend/Stockr.Domain/Entities/Product.cs
namespace Stockr.Domain.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; }
    public string? SKU { get; set; }
    public string Description { get; set; }
    public Guid CategoryId { get; set; }
    public Category Category { get; set; }
    public Guid? SupplierId { get; set; }
    public Supplier? Supplier { get; set; }
    public decimal CostPrice { get; set; }
    public decimal SalePrice { get; set; }
    public Guid? TenantId { get; set; }
    public Tenant? Tenant { get; set; }
}